### PR TITLE
Updating displayName Reactive #2994

### PR DESCRIPTION
**Pull request by DeNic0la**
PR for issue #2994

Reload user entity after changing `firstname`, `surname `or `nickname`

### CODE REVIEW

**Comment by manuelmeister** [comment]
So cool. Thank you for your contribution. I'll review it tomorrow 😊

File: frontend/src/views/Profile.vue [hunk 1]
```diff
@@ -119,6 +122,13 @@ export default {
       this.api.reload(this.user).then((user) => this.api.reload(user.profile()))
     }
   },
+  methods: {
+    reloadUser() {
+      this.api.reload(this.user).then((user) => {
+        this.$store.commit('login', user)
```
Comment (carlobeltrame) [inline frontend/src/views/Profile.vue]: To make this more easily understandable, could you add another mutation called "updateUser" to
frontend/src/components/store/auth.js, and then use that here?
The mutation code (single line) will be duplicated, but what we're doing here isn't really a re-login, but an update of the user state. Maybe these two actions will have different behaviour in the future.

**Comment by carlobeltrame** [comment]
I also noticed the update is not applied if I navigate away from the profile edit during the save operation (I was testing on my phone). Not sure how to solve this though...

**Comment by carlobeltrame** [comment]
Maybe it was also a bad idea of us in the first place, to store a duplicated user state in the auth store. Maybe we would be better off storing only the user URI (which we get from the JWT from the API) there, and reading the user data from the normal API store when needed. But this `auth.user` is used in [too many places](https://github.com/search?q=repo%3Aecamp%2Fecamp3%20auth.user&type=code) so you don't have to refactor this in this PR.

**Review by carlobeltrame — APPROVED** [review]
Thanks!

**Review by pmattmann — APPROVED** [review]

**Comment by pmattmann** [comment]
@DeNic0la: Very cool - thx!

**Pull request merged**